[HOPSWORKS-3328] Improve documentation around feature group partition…

SirOibaf opened the pull request:
… keys

shanksy01 reviewed (changes requested):
2 comments:

1. Is there a negative connotation to partitioning (having too many partitions) ie can you only have 1 partition per feature group?
2. You mentioned in the case that partitioning can help query times - it would be good to mention that as well

moritzmeister reviewed (commented):
Should we add that a good practice is to use a coarse grained form of the event time stamp as partition key?

Should we add that adding a partition key such that when inserting new data, old data won't be touched is a good idea?

SirOibaf commented:
Addressed the feedback. Also added information about the requirements/behavior for primary keys/event time.

@shanksy01 - Having a single partition will have the equivalent effect has having no partition at all as all the data will be stored in a single directory

@moritzmeister - I don't think you should be able to change the partition keys after a feature group was created.

shanksy01 reviewed (changes requested):
typo line 84 (add a "t"): The even time >> The event time

typo line 88 (add a comma): When you specify a partition key the data >> When you specify a partition key, the data 

Otherwise spot on! Nice job.
